source/extensions/filters/udp/dns_filter/dns_filter.cc

```diff
@@ -39,9 +39,11 @@ DnsFilterEnvoyConfig::DnsFilterEnvoyConfig(
   for (const auto& virtual_domain : dns_table.virtual_domains()) {
     AddressConstPtrVec addrs{};
 
-    const absl::string_view domain_name = virtual_domain.name();
-    const absl::string_view suffix = Utils::getDomainSuffix(domain_name);
-    ENVOY_LOG(trace, "Loading configuration for domain: {}. Suffix: {}", domain_name, suffix);
+    const absl::string_view virtual_domain_name =
+        Utils::getVirtualDomainName(virtual_domain.name());
+    const absl::string_view suffix = Utils::getDomainSuffix(virtual_domain_name);
```

Comment:
Should we getDomainSuffix(virtual_domain.name()),
 or  getDomainSuffix( Utils::getVirtualDomainName(virtual_domain.name()))

Can these two return different result in some cases?

Reply:
Yeah, I thought about this (getting suffix of "initial" input domain name). But with current implementation result should always be the same.